src/issue-creator/cb-open-issue-creator.ts

```diff
@@ -0,0 +1,26 @@
+import { HttpService } from '@nestjs/common';
+import { LogMessageFormat, LogType } from 'logging-format';
+import { IssueCreator } from './issue-creator';
+import { ConfigService } from '@nestjs/config';
+
+/**
+ * CbOpenIssueComponent handles CB open Logs, it extends IssueCreator to enable individual issue creation for cb open issues 
+ */
+export class CbOpenIssueCreatorComponent extends IssueCreator {
+  issueCreator: IssueCreator;
+  constructor(http: HttpService, configService: ConfigService) {
+    super(http, configService);
+    this.issueCreator = new IssueCreator(http, configService);
+  }
+
+  /**
+   * handles cp open logs by creating an Issue and sending it to the API: https://github.com/ccims/ccims-backend/tree/apiMockup
+   * 
+   * @param log received log 
+   * @returns the issue ID received from the backend
+   */
+  async handleLog(log: LogMessageFormat) {
+    if (log.type != LogType.CB_OPEN) throw 'Wrong LogType';
+    return this.issueCreator.createIssueFromLog(log);
+  }
+}
```

Comment:
@Jacky20000 ich glaube, du hast das konzept der vererbung nicht verstanden. ich empfehle dir, dich nochmal mit den PSE unterlagen auseinander zusetzen.

das tolle an vererbung ist schließlich, dass kinder die methoden und eigentschaften ihrere eltern erben - auch in typescript ([https://www.typescriptlang.org/docs/handbook/classes.html#inheritance](https://www.typescriptlang.org/docs/handbook/classes.html#inheritance))

Reply:
hierüber bitte noch nachdenken und meinen vorschlag entweder einarbeiten, oder begründen warum es ein doofer vorschlag ist.